### PR TITLE
add Posts feature page and required cms components

### PR DESCRIPTION
**Pull request by kaitoo1**
### Summary of Changes
This PR adds a Feature Page to introduce posts. it can be found at /features/posts.
The content is provided via our CMS, and rendered using new components that match the CMS content models.

### Demo or Before/After Pics
CleanShot 2023-10-13 at 13 37 10

CleanShot 2023-10-13 at 13 37 06

CleanShot 2023-10-13 at 13 36 56

CleanShot 2023-10-13 at 13 36 59

### Edge Cases
- mobile responsive

### Testing Steps
- go to /features/posts

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/pages/features/posts.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pageContent: CmsTypes.FeaturePage;
+};
+
+export default function PostsFeatureRoute({ pageContent }: Props) {
+  return <GalleryRoute element={<PostsFeaturePage pageContent={pageContent} />} navbar={false} />;
+}
+
+export const getServerSideProps = async () => {
+  const query = `
+  *[ _type == "featurePage" && id == "posts" ]{
```
Comment (Robinnnnn) [inline apps/web/pages/features/posts.tsx:16]: at first i was wondering what kind of graphql fuckery this was

File: apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+type Props = {
+  content: CmsTypes.FaqModule;
+};
+
+export default function Faq({ content }: Props) {
+  const [openIndex, setOpenIndex] = useState<number | null>(null);
+
+  const handleClick = (index: number) => {
+    setOpenIndex(openIndex === index ? null : index);
```
Comment (Robinnnnn) [inline apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx:126]: nit: you should just be able to do `setOpenIndex(index)`. AFAIK react should know that nothing has changed

File: apps/web/src/scenes/ContentPages/ContentModules/FeatureHighlight.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+import { CmsTypes } from '../cms_types';
+
+type FeatureHighlightBulletsProps = {
+  bullets: CmsTypes.FeatureHighlight['body'];
+};
+
+function FeatureHighlightBullets({ bullets }: FeatureHighlightBulletsProps) {
+  const textList = useMemo(
+    () => bullets.map((bullet) => bullet.children[0] && bullet.children[0].text),
```
Comment (Robinnnnn) [inline apps/web/src/scenes/ContentPages/ContentModules/FeatureHighlight.tsx:16]: supernit: `bullet.children[0]?.text`

File: apps/web/src/scenes/ContentPages/PostsFeaturePage.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+type Props = {
+  pageContent: CmsTypes.FeaturePage;
+};
+
+export default function PostsFeaturePage({ pageContent }: Props) {
+  const track = useTrack();
+  const handleGetStartedClick = useCallback(() => {
+    track('Posts Feature Page: Clicked Get Started');
+  }, [track]);
```
Comment (Robinnnnn) [inline apps/web/src/scenes/ContentPages/PostsFeaturePage.tsx:25]: this will be done directly on the button soon :)

**Review by Robinnnnn — APPROVED** [review]

File: apps/web/pages/features/posts.tsx (hunk 1 above)
Comment (kaitoo1) [reply apps/web/pages/features/posts.tsx:16]: haha its sanity's query language GROQ which conceptually is very similar

File: apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx (hunk 2 above)
Comment (kaitoo1) [reply apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx]: sweet, simplified!

File: apps/web/src/scenes/ContentPages/ContentModules/FeatureHighlight.tsx (hunk 3 above)
Comment (kaitoo1) [reply apps/web/src/scenes/ContentPages/ContentModules/FeatureHighlight.tsx]: great call

File: apps/web/src/scenes/ContentPages/PostsFeaturePage.tsx (hunk 4 above)
Comment (kaitoo1) [reply apps/web/src/scenes/ContentPages/PostsFeaturePage.tsx:25]: yess so amazing

**Review by Robinnnnn — APPROVED** [review]

File: apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx (hunk 2 above)
Comment (kaitoo1) [reply apps/web/src/scenes/ContentPages/ContentModules/Faq.tsx:128]: nvm remembered I needed this in order to close a section that's currently open! so I added it back

**Pull request merged**